src/big/big_int.cr

```diff
@@ -503,11 +503,7 @@ struct BigInt < Int
   end
 
   def to_i64!
-    if LibGMP::Long == Int64 || (Int32::MIN <= self <= Int32::MAX)
-      LibGMP.get_si(self).to_i64!
-    else
-      to_s.to_i64
-    end
+    (self % BITS64).to_u64.to_i64!
```

Comment:
Why not reuse `to_u64!`?

```suggestion
    to_u64!.to_i64!
```

Reply:
Right... this is a leftover from an earlier attempt / workaround

Reply:
Oh I see what you mean. Hmm actually no, I don't like it. The code happens to match, but I think this better shows the intent that this roundtrip is intentional.